lucet-runtime/lucet-runtime-tests/src/guest_fault.rs

```diff
@@ -563,53 +599,54 @@ macro_rules! guest_fault_tests {
+                    let recoverable_ptr_lock = RECOVERABLE_PTR_LOCK.lock().unwrap();
+                    let module = mock_traps_module();
+                    let region =
+                        TestRegion::create(1, &Limits::default()).expect("region can be created");
+                    let mut inst = region
+                        .new_instance(module)
+                        .expect("instance can be created");
 
-                match inst.run("illegal_instr", &[]) {
-                    Err(Error::RuntimeFault(details)) => {
-                        assert_eq!(details.trapcode, Some(TrapCode::BadSignature));
+                    let sa = SigAction::new(
+                        SigHandler::SigAction(host_sigsegv_handler),
+                        SaFlags::SA_RESTART,
+                        SigSet::all(),
+                    );
+                    let before_sa = unsafe {
+                        sigaction(INVALID_PERMISSION_SIGNAL, &sa).expect("sigaction succeeds")
+                    };
+
+                    match inst.run("illegal_instr", &[]) {
+                        Err(Error::RuntimeFault(details)) => {
+                            assert_eq!(details.trapcode, Some(TrapCode::BadSignature));
+                        }
+                        res => panic!("unexpected result: {:?}", res),
                     }
-                    res => panic!("unexpected result: {:?}", res),
-                }
 
-                // now make sure that the host sigaction has been restored
-                unsafe {
-                    recoverable_ptr_setup();
-                }
-                *HOST_FAULT_TRIGGERED.lock().unwrap() = false;
+                    // now make sure that the host sigaction has been restored
+                    unsafe {
+                        recoverable_ptr_setup();
+                    }
+                    *HOST_FAULT_TRIGGERED.lock().unwrap() = false;
 
-                // accessing this should trigger the segfault
-                unsafe {
-                    *RECOVERABLE_PTR = 0;
-                }
+                    // accessing this should trigger the segfault
+                    unsafe {
+                        *RECOVERABLE_PTR = 0;
+                    }
 
-                assert!(*HOST_FAULT_TRIGGERED.lock().unwrap());
+                    assert!(*HOST_FAULT_TRIGGERED.lock().unwrap());
 
-                // clean up
-                unsafe {
-                    recoverable_ptr_teardown();
-                    sigaction(
-                        INVALID_PERMISSION_SIGNAL,
-                        &SigAction::new(SigHandler::SigDfl, SaFlags::SA_RESTART, SigSet::empty()),
-                    )
-                    .expect("sigaction succeeds");
-                }
+                    // clean up
+                    unsafe {
+                        recoverable_ptr_teardown();
+                        sigaction(INVALID_PERMISSION_SIGNAL, &before_sa)
```

Comment:
:bug: 

Turns out `SigDfl` is not the same as the handler Rust installs for stack overflows.